Framework/Kernel/src/Matrix.cpp

```diff
@@ -110,32 +114,32 @@ Matrix<T>::Matrix(const std::vector<T> &A, const std::vector<T> &B)
   @param B :: Row vector to multiply
 */
 {
-  // Note:: nx,ny zeroed so setMem always works
+  // Note:: m_numRowsX,m_numColumnsY zeroed so setMem always works
   setMem(A.size(), B.size());
-  for (size_t i = 0; i < nx; i++) {
-    for (size_t j = 0; j < ny; j++) {
-      V[i][j] = A[i] * B[j];
+  for (size_t i = 0; i < m_numRowsX; i++) {
+    for (size_t j = 0; j < m_numColumnsY; j++) {
+      m_rawData[i][j] = A[i] * B[j];
     }
   }
 }
 //
 template <typename T>
 Matrix<T>::Matrix(const std::vector<T> &data)
-    : nx(0), ny(0), V(nullptr) {
+    : m_numRowsX(0), m_numColumnsY(0) {
   size_t numel = data.size();
-  size_t nxt = static_cast<size_t>(sqrt(double(numel)));
-  size_t test = nxt * nxt;
+  size_t m_numRowsXt = static_cast<size_t>(sqrt(double(numel)));
```

Comment:
But we throw if it's not valid so if it's not square there won't be any object anyway...

Wouldn't something like:

```cpp
size_t numElements= data.size();
m_numRows = static_cast<size_t>(sqrt(double(numElements)));
if (numElements != m_numRows*m_numRows ) {
  throw(std::invalid_argument("...."
}
setMem(m_numRows , m_numRows);
...
```

work?